### PR TITLE
Add total requests prometheus metric

### LINKED ISSUE #415: Additional prometheus metrics
Additional prometheus metrics would be helpful for making decisions in the future.
A concrete example is exporting the called routes, which would be useful for deprecating endpoints as in #332.

### PR DESCRIPTION
**Pull request by M4tthewDE**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Adds an additional metric which tracks which top level routes are used how often. Not quiet happy with how the URL parsing works, but I was not able to find a better way with the std lib.

Closes #415

### CODE REVIEW

**Comment by pajlada** [comment]
Could you provide an example of what data is tracked? (i.e. what it looks like at the prometheus endpoint)

**Comment by M4tthewDE** [comment]
I switched to the premade middleware, since it has better URL parsing and more detailed metrics.

How the metrics look like: https://github.com/766b/chi-prometheus#what-do-you-get

File: cmd/api/prometheus.go [hunk 1]
```diff
@@ -5,9 +5,19 @@ import (
...
 )
 
+var (
+	servedRoutes = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "requests_total",
+		Help: "Total number of requests",
+	},
+		[]string{"path"},
+	)
+)
```
Comment (pajlada) [inline cmd/api/prometheus.go:19]: This doesn't seem to be used, can this be removed?

**Review by pajlada — CHANGES_REQUESTED** [review]

File: cmd/api/prometheus.go (hunk 1 above)
Comment (M4tthewDE) [reply cmd/api/prometheus.go:19]: that is indeed unneeded

**Review by pajlada — APPROVED** [review]

**Pull request merged**